### ISSUE TITLE
Assert  $exception->getMessages()  not working properly 

### ISSUE THREAD

**Issue by hondnl** [Respect/Validation#446]
When using the example Getting Messages from the feature guide you will get an empty array when only having 1 error message.

Example :

```php
$arr=array("w","hello@hello.com");//invalid first key only...

try {
    v::key('name', v::length(2, 32))
            ->key('email', v::email())
            ->assert($arr);
} catch(NestedValidationExceptionInterface $e) {
    print_r($e->getMessages());
}
```

Will give you an empty array....

Reason and hotfix :  
Exceptions/AbstractNestedException.php
Method :getMessages 

In the for loop it removes the first key .. When only 1 invalid key , it removes the error message needed.
Fix : something like this.. or something cleaner further up the class.

```php
public function getMessages()
{
    $messages = array();

    foreach ($this->getIterator() as $key => $exception) {
        $messages[] = $exception->getMessage();
    }

    if(count($messages) >1)
        array_shift($messages);

    return $messages;
}
```

**Comment by henriquemoody** [comment]
I took a look here, that's not the problem of the `getMessages()` method, this is the same problem described on #179, you can see that when using the `getFullMessage()` method.

I'm going to fix this the next week.

I'm closing it because it's duplicated. Thanks for reporting! :panda_face:

**Comment by henriquemoody** [comment]
This fix will be released on Saturday, Oct 24th, with version 1.0:

This test was created to ensure this is working: https://github.com/Respect/Validation/blob/master/tests/integration/issue-446.phpt
